### PR TITLE
Improve cron error logging

### LINKED ISSUE #37453: [Issue] Improve cron error logging
This issue is automatically created based on existing pull request: magento/magento2#32690: Improve cron error logging

---------
### Description (*)
This pull request improves logging of errors within cron processes.

**Current scenario**
Currently any error messages are lost / not recorded. For the main group (and any groups not set to use a separate process), STDERR is left for the scheduler to capture. This is most easily caught with the `MAILTO=` directive on unix, however experience suggests this is rarely configured usefully. Any cron group currently configured to run in a separate process will have its STDOUT and STDERR lost forever.

**Proposed scenario**
With the changes here, both STDERR and STDOUT are caught and logged. This is very valuable in scenarios where one cannot configure the scheduler, or when a job running in a separate process is suffering from errors. A good example of this is Magento Cloud hosting where cron processes often fail without any useful diagnostic information available. These changes will mean that STDERR output is readily available, aiding in diagnostics.

### Fixed Issues (if relevant)
*None*

### Manual testing scenarios (*)
1. Artificially introduce a memory leak into a cron job, eg `indexer_reindex_all_invalid`
1. Run `php bin/magento cron:run` without these changes
1. Observe lack of information about the failure
1. Run `php bin/magento cron:run` with these changes
1. Observe information available in `var/log` (in this example, `var/log/magento.cron.index.log`)

### Questions or comments
I can see that [using `posix_isatty()` is discouraged](https://github.com/magento/magento-coding-standard/blob/887ecba8edaff155951094051822995cff3bca1c/Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php#L107), but no alternative is provided / suggested. There are functions in packages installed via composer as development dependencies which provide the required functionality. (eg [here](https://github.com/sebastianbergmann/environment/blob/5.1.3/src/Console.php#L105) and [here](https://github.com/squizlabs/PHP_CodeSniffer/blob/3.5.8/src/Util/Common.php#L160)) What's the recommended approach in this case? Do we simply remove the output line altogether?

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by fredden**
### Description (*)
This pull request improves logging of errors within cron processes.

**Current scenario**
Currently any error messages are lost / not recorded. For the main group (and any groups not set to use a separate process), STDERR is left for the scheduler to capture. This is most easily caught with the `MAILTO=` directive on unix, however experience suggests this is rarely configured usefully. Any cron group currently configured to run in a separate process will have its STDOUT and STDERR lost forever.

**Proposed scenario**
With the changes here, both STDERR and STDOUT are caught and logged. This is very valuable in scenarios where one cannot configure the scheduler, or when a job running in a separate process is suffering from errors. A good example of this is Magento Cloud hosting where cron processes often fail without any useful diagnostic information available. These changes will mean that STDERR output is readily available, aiding in diagnostics.

### Fixed Issues (if relevant)
*None*

### Manual testing scenarios (*)
1. Artificially introduce a memory leak into a cron job, eg `indexer_reindex_all_invalid`
1. Run `php bin/magento cron:run` without these changes
1. Observe lack of information about the failure
1. Run `php bin/magento cron:run` with these changes
1. Observe information available in `var/log` (in this example, `var/log/magento.cron.index.log`)

### Questions or comments
I can see that [using `posix_isatty()` is discouraged](https://github.com/magento/magento-coding-standard/blob/887ecba8edaff155951094051822995cff3bca1c/Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php#L107), but no alternative is provided / suggested. There are functions in packages installed via composer as development dependencies which provide the required functionality. (eg [here](https://github.com/sebastianbergmann/environment/blob/5.1.3/src/Console.php#L105) and [here](https://github.com/squizlabs/PHP_CodeSniffer/blob/3.5.8/src/Util/Common.php#L160)) What's the recommended approach in this case? Do we simply remove the output line altogether?

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#37453: Improve cron error logging

### CODE REVIEW

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by evktalo** [comment]
I can confirm this issue has caused a problem with monitoring for one of our clients.

**Comment by fredden** [comment]
> I can confirm this issue has caused a problem with monitoring for one of our clients.

@evktalo I'm having a little trouble interpreting your statement. Are you saying that (1) you have applied the changes in this pull request to a website and these have caused a problem somehow? Or, are you saying that (2) you have witnessed the issue that this pull request aims to solve?

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
Hello @fredden,

Thanks for the contribution!

We have tried to reproduce the issue in the latest 2.4-develop branch, but it seems the issue is no more reproducible. As per the steps mentioned here https://github.com/magento/magento2/pull/32690#issue-852871342, we have tried to artificially introduce a memory leak into a cron job in the below file:

https://github.com/magento/magento2/blob/9c0e1236743f5cccb13cfbf72cf1473aaebd2d9c/app/code/Magento/Indexer/Cron/ReindexAllInvalid.php#L31

And tried to add the below code to introduce a memory leak:

```
$leak = [];
        $ticks = 0;
        while(true) {
            $leak[$ticks] =  "{A VERY LONG STRING}";
            $ticks++;
            unset($leak[$ticks]);
        }
```

But the issue is not reproducible for us. Please let us know if we have missed anything.

Thanks

**Comment by engcom-Hotel** [comment]
Hello @fredden,

I think you are talking about the below error:

image

I am able to generate the error after setting up the lower memory_limit.

Thanks

**Comment by engcom-Hotel** [comment]
Hello @fredden,

We have tried this PR to check the error logs with the same memory leak codebase. This time with this PR changes, we have not gotten any error as mentioned in this https://github.com/magento/magento2/pull/32690#issuecomment-1521680879. 

But either not getting any error log in the log file as well:

image

let us know if we have missed anything.

Thanks

**Comment by fredden** [comment]
@engcom-Hotel what log files are you looking in? The snippet you posted seems to be from `var/log/cron.log`. Have you checked other log files, such as `var/log/magento.cron.index.log` or `var/log/magento.cron.consumers.log`?

I've tested this just now by adding the following to `bin/magento`:

```diff
diff --git a/bin/magento b/bin/magento
index 0a5f498dd7e..d4ffb28aa6c 100755
--- a/bin/magento
+++ b/bin/magento
@@ -10,6 +10,11 @@ if (PHP_SAPI !== 'cli') {
     exit(1);
 }
 
+fwrite(STDOUT, "This string has been written to STDOUT\n");
+fwrite(STDERR, "This string has been written to STDERR\n");
+
+print_r($argv);
+
 try {
     require __DIR__ . '/../app/bootstrap.php';
 } catch (\Exception $e) {
```

When I run `php bin/magento cron:run`, I get expected output:

Screenshot_2023-04-25_15-54-14

I also get the PHP Fatal error logged as expected when I introduce a memory leak as suggested:

Screenshot_2023-04-25_15-57-58

**Comment by fredden** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
Hello @fredden,

I have just rechecked the issue and found the logs in `var/log/magento.cron.index.log`:

image

Thanks

**Comment by fredden** [comment]
The test failures from "Functional Tests B2B", "Functional Tests CE", "Functional Tests EE", and "Integration Tests" are all out of scope for this pull request. I have not done any research to determine if these tests are unreliable or broken elsewhere. They are not failing due to any changes introduced in this pull request.

There are two errors showing in the "Static Test" suite. One error (reported under a heading of "Php (B2B)") is already covered in the pull request description above. The other error (reported for both "Integrity (B2B)" and "Integrity (EE)") will need to be resolved by someone who has access to the now-private repositories for "enterprise edition" (ie, Adobe Commerce) and "business to business" (which isn't an edition).

I have fixed the Unit Test failure in 365236a125d7f9fc35aa6f683dca3b8954a0fe3a.

**Review by ihor-sviziev — APPROVED** [review]

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please fix failing static tests, all other failures look not relared

**Comment by fredden** [comment]
> Please fix failing static tests, all other failures look not related

Would you like me to add a 'phpcs:ignore' entry for the use of `posix_isatty()`? See note in pull request description. I am unable to influence the other static test failures as I do not have access to the relevant repositories on GitHub.

**Comment by fredden** [comment]
I have added a1d3859b6c2146b6fd24a818ae3f8b8ac1756a44 to ignore the use of `posix_isatty()`.

**Comment by fredden** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by engcom-Hotel** [comment]
@fredden @ihor-sviziev we have created a guild issue to discuss on the usage of `posix` function in this PR. You can refer to the same here https://github.com/magento-commerce/development-guild/issues/202.

I will update you the discussion outcome here soon.

Thanks

**Comment by fredden** [comment]
@engcom-Hotel that URL does not work for me (because the repository is private). Do you expect us to join the conversation, or are you telling us that there is a private-to-Adobe discussion taking place? Why not have the discussion here?

**Comment by engcom-Hotel** [comment]
Hello @fredden,

We are in discussion with the internal team through Guild for using the POSIX extension, hence we are moving this PR `on hold` status. 

We will proceed further once we get the conclusion from the internal team.

Thanks

File: app/code/Magento/Cron/Console/Command/CronCommand.php
Comment (engcom-Hotel) [inline app/code/Magento/Cron/Console/Command/CronCommand.php]: Hello @fredden,

just checking the documentation of [stream_isatty](https://www.php.net/manual/en/function.stream-isatty.php),  can we replace `posix_isatty` with it.

File: lib/internal/Magento/Framework/Shell/CommandRendererBackground.php
Comment (engcom-Hotel) [inline lib/internal/Magento/Framework/Shell/CommandRendererBackground.php]: As per discussion in the guild, $arguments[2] is not supposed to $group_id always, since it’s a general class that might be used not only by cron:run command.

File: app/code/Magento/Cron/Console/Command/CronCommand.php
Comment (fredden) [reply app/code/Magento/Cron/Console/Command/CronCommand.php]: I've made this change. `stream_isatty()` is also a "discouraged" function.

File: lib/internal/Magento/Framework/Shell/CommandRendererBackground.php
Comment (fredden) [reply lib/internal/Magento/Framework/Shell/CommandRendererBackground.php]: Based on this, I have removed the modifications to the Framework classes.

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

File: app/code/Magento/MessageQueue/etc/di.xml
Comment (ihor-sviziev) [inline app/code/Magento/MessageQueue/etc/di.xml]: According to SVC failure, I guess we should move this part back to the CRON module, and maybe as a separate task move it to another. 
BTW, it looks like it should be in global di.xml which is located in app/etc, not in message queue module.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/MessageQueue/etc/di.xml
Comment (ihor-sviziev) [reply app/code/Magento/MessageQueue/etc/di.xml]: Also there some functional test failures related to crons, are you sure you didn't broke anything?

Comment (fredden) [reply app/code/Magento/MessageQueue/etc/di.xml]: I've moved this to `app/etc/di.xml` as suggested. (It only appears to be used in `app/code/Magento/MessageQueue/etc/di.xml` as `Magento_Cron` now uses a different `virtualType`.)

I don't see which functional test failures you're referring to; can you share a link or explain what you're seeing? The functional test failures look out of scope for this pull request.

**Comment by fredden** [comment]
@magento run all tests

File: app/code/Magento/MessageQueue/etc/di.xml
Comment (ihor-sviziev) [reply app/code/Magento/MessageQueue/etc/di.xml]: From structure perpective it's the right place, but it causing SVC build failure, that has to be approved. Unfortunately, I never saw fast approvals on such failures :( 
My suggestion - move it back to Cron module for now, and maybe in a separate PR moving it to app/etc/di.xml.
What do you think?

About the test: here is https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32690/0c35aa78e709b4cfc79028105f41ee6f/Functional/allure-report-b2b/index.html#categories/3260ddfd49d84bde7dc699a2b3e199ea/7a8e823f1224a39/
It's named "Verify Cron Partial Reindex for Products assigned to Categories and several new websites". I don't see direct reason that it fails on some command, but some product do not appeared, so it might be related to changes in crons.

Comment (fredden) [reply app/code/Magento/MessageQueue/etc/di.xml]: The "Semantic Version Checker" test failure is a false positive. The tool should be fixed. This pull request has already been open for more than two years.

From what I can tell, the failed functional test that you have linked is a race condition / poorly designed test. I will try to adjust the test now to ensure it works reliably.
Edit: I have opened https://github.com/magento/magento2/pull/37511 to improve the reliability of that test.

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests look not related to changes from this PR.

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Lima** [comment]
Hi @fredden ,

Thanks for your contribution and collaboration.

PR is not working as expected, after taking PR changes It is observed that logs are not writing in the `magento.cron.index.log` file of memory limit. 

Below are the steps which I have followed:
- Reduce the memory limit in the Php.ini to 2M.
- Add code in `/bin/magento` file

```
fwrite(STDOUT, "This string has been written to STDOUT\n");
fwrite(STDERR, "This string has been written to STDERR\n");

print_r($argv);
```
- Run `php bin/magento cron:run`
- Observe the log in  `cat magento.cron.index.log`

**Below is the screenshot for your reference:**

image

image

image

Kindly recheck it and provide us information if we are missing anything.

Thanks

**Comment by ihor-sviziev** [comment]
@engcom-Lima, the issue is actually opposite, before PR it wasn't written to logs, while now it should be.

**Comment by engcom-Lima** [comment]
Hi @ihor-sviziev 

When I have tested it earlier, error were logged in the logs(After taking PR changes).
Please refer my [previous comment](https://github.com/magento/magento2/pull/32690#issuecomment-1532596107).

Thanks

**Comment by fredden** [comment]
@engcom-Lima I suspect that you have set the memory limit too low for Magento to properly initialise. Do any cron jobs run at all with a memory limit of 2 MiB?
I recommend setting the memory limit to be `2G` and introducing a memory leak into a cron job which gets run every minute - eg `indexer_reindex_all_invalid` whose code is in `app/code/Magento/Indexer/Cron/ReindexAllInvalid.php`.
You may like to discuss this with @engcom-Hotel who has successfully tested this previously.

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**

**Manual testing scenario:**
- Reduce the memory limit in the `php.ini`.
- Add below code in `app/code/Magento/Indexer/Cron/ReindexAllInvalid.php`
```
$leak = [];
        $ticks = 0;
        while(true) {
            $leak[$ticks] =  "{A VERY LONG STRING}";
            $ticks++;
            unset($leak[$ticks]);
        } 
```
- Add code in `/bin/magento` file.

```
fwrite(STDOUT, "This string has been written to STDOUT\n");
fwrite(STDERR, "This string has been written to STDERR\n");

print_r($argv);
```
- Run `php bin/magento cron:run`
- Observe the log in `cat magento.cron.index.log`

**After:** :heavy_check_mark:  
Screenshot 2023-05-19 at 1 33 09 PM

**Comment by engcom-Charlie** [comment]
It seems that bot has moved this PR in Ready for testing from Extended testing hence moved it back.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
Hi @fredden,

Thank you for your contribution!

Can you please add the changes from https://github.com/magento/magento2/pull/37511 to this PR. In order to move PR further, we need all tests should get passed so it will be great if you club that in to this PR.

Thank you!

**Comment by fredden** [comment]
No, that pull request is separate from this on purpose. The scope of this pull request is to improve the logging that the 'run:cron' process produces. The linked pull request intends to improve the reliability of a particular test in the functional testing suite. While these are related, they are intentionally distinct and separate.

It sounds like you want to put this pull request on hold until the testing suite is (more) reliable. There are many more unreliable tests in the various testing suites.

The changes in the other pull request do not affect the changes here, and the inverse is also true. As far as I know, the automated test failures showing in this pull request are out of scope here. Please let me know if you require me to investigate a particular test failure in detail for its relevance to this pull request.

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE

File: app/code/Magento/Catalog/Test/Mftf/Test/StorefrontVerifyProductAfterPartialReindexOnSeveralWebsitesTest.xml
Comment (fredden) [inline app/code/Magento/Catalog/Test/Mftf/Test/StorefrontVerifyProductAfterPartialReindexOnSeveralWebsitesTest.xml]: @engcom-Charlie this change is out of scope for this pull request. Please can you move this to a separate pull request.

Edit: this looks like a duplicate of https://github.com/magento/magento2/pull/37511, which I specifically said is separate from this pull request on purpose. See https://github.com/magento/magento2/pull/32690#issuecomment-1558671953

**Comment by fredden** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
Hi @fredden,

As part of the process, each PR's build needs to be green to proceed further or else the same tests should not fail on multiple rerun.

Currently StorefrontVerifyProductAfterPartialReindexOnSeveralWebsitesTest is failing on each run so we need to analyse it that is it failing because of this PR changes or not. 

As this test is working on my local and not on jenkins, currently as you can see I have added it as draft commit only. Once we get the analysis done for this we will revert it back or do the needful. 

Thank you!

**Comment by engcom-Charlie** [comment]
As bot moved it to Review, moving it back Extended Testing.

Thank you!

**Comment by fredden** [comment]
> As part of the process, each PR's build needs to be green to proceed further or else the same tests should not fail on multiple rerun.

It would be nice if that same process applied to changes made by Adobe internally. This sounds like a sensible approach. When tests are identified as unreliable, are these recorded somewhere so they can be eventually fixed? Is that list publicly available? It seems like a good place where the community could contribute.

> Currently StorefrontVerifyProductAfterPartialReindexOnSeveralWebsitesTest is failing on each run so we need to analyse it that is it failing because of this PR changes or not.

I have already assessed this test, and determined that the reason for failure is out of scope for this pull request. I opened https://github.com/magento/magento2/pull/37511 to fix the unreliable test. See https://github.com/magento/magento2/pull/32690#discussion_r1196637387. If you require this test to pass consistently, I recommend you review https://github.com/magento/magento2/pull/37511 and get that merged. You are of course welcome to review the test and the reason for its failure yourself.

Please let me know if you think that further changes should be made to this pull request.

**Comment by fredden** [comment]
Thanks @ihor-sviziev. I've run https://github.com/magento/magento-semver over the changes here. It reports that the changes are 'patch', so there must be a different tool that Adobe are running for their "Semantic Version Checker" check on GitHub.

**Comment by ihor-sviziev** [comment]
@fredden, I just checked the same repo - it shows for me that major changes detected:
image
I used the following instruction: https://developer.adobe.com/commerce/testing/guide/svc/
Note: you should have 2 Magento installations - one is a clean 2.4-develop branch, and 2nd one is a branch with your changes.

**Comment by engcom-Charlie** [comment]
Hi @fredden,

For Semantic Version Checker failure, in this case as per the process we will create the approval jira ticket so you ignore that.

About Integration test failure, its not related to this PR. 

For the remaining functional test failures, we are looking into it.

Thank you!

**Comment by fredden** [comment]
> I just checked the same repo - it shows for me that major changes detected

Thanks for checking this. I've run this again now over the whole Magento2 codebase and the tool is reporting the same as your screen-shot. Previously I ran this on the module only. (I gave up waiting for the whole-code-base analysis to complete; today I let it run for more than five minutes and it completed.)

> For Semantic Version Checker failure, in this case as per the process we will create the approval jira ticket so you ignore that.

This is different from the process that you advised previously. Is there a public document describing "the process"? It's very confusing from this side of the Adobe wall. I can't tell what test failures you'll accept and which you'll block for.

**Comment by engcom-Charlie** [comment]
> 

Process is same as I have mentioned in the [comment](https://github.com/magento/magento2/pull/32690#issuecomment-1561281257).

For some specific exceptional cases, because of the PR implementation when we got Magento Health index, Database compare, Semantic version checker etc failure then after the analysis we have to raise a internal JIRA approval ticket.

Thank you!

**Comment by engcom-Charlie** [comment]
> > I just checked the same repo - it shows for me that major changes detected
> 
> Thanks for checking this. I've run this again now over the whole Magento2 codebase and the tool is reporting the same as your screen-shot. Previously I ran this on the module only. (I gave up waiting for the whole-code-base analysis to complete; today I let it run for more than five minutes and it completed.)
> 
> > For Semantic Version Checker failure, in this case as per the process we will create the approval jira ticket so you ignore that.
> 
> This is different from [the process that you advised previously](https://github.com/magento/magento2/pull/32690#issuecomment-1561281257). Is there a public document describing "the process"? It's very confusing from this side of the Adobe wall. I can't tell what test failures you'll accept and which you'll block for.

Process is same as I have mentioned in the [comment](https://github.com/magento/magento2/pull/32690#issuecomment-1561281257).

But for some specific exceptional cases, because of the PR implementation when we got Magento Health index, Database compare, Semantic version checker etc failure then after the analysis we have to raise a internal JIRA approval ticket.

Thank you!

**Comment by fredden** [comment]
I have opened https://github.com/magento/magento-semver/pull/72 to resolve the test failure for Semantic Version Checker.

**Comment by fredden** [comment]
Thanks @engcom-Charlie. Yes, that sounds like a good summary.

I have concluded my investigation and can confirm that the test failures are out of scope here. Pull requests exist to resolve the issues identified:
https://github.com/magento/magento2/pull/37511
https://github.com/magento/magento-semver/pull/72
https://github.com/magento/magento2-functional-testing-framework/pull/904

When these three are merged in, we should be able to progress with this pull request. Or if you would prefer, you can apply for the necessary concessions internally at Adobe to ignore these test failures and merge this as-is. I do not believe any further changes to this pull request are required.

**Comment by engcom-Charlie** [comment]
Hi @fredden,

Thank you for this reply.

We greatly appreciate your contribution!!

Thank you for providing the implementation in below PRs

1. https://github.com/magento/magento2/pull/37511
2. https://github.com/magento/magento2-functional-testing-framework/pull/904

As you mentioned, these are the some modifications/enhancements you have done in order to take care of few failures (eg. like failures in this PR). Hence, we will first look into and process above PRs. Once they will get merge on 2.4 develop we will proceed on this PR accordingly.

Till then moving this one to On hold.

Thank you once again!!

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

**Review by andrewbess — APPROVED** [review]
Hello @fredden 
Thank you for your contribution. This is great job.
It looks good for me.
Waiting for approval ticket.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
As this PR has been moved from Extended Testing to On hold, now moving back it to Extended Testing.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
As mentioned [here](https://github.com/magento/magento2/pull/32690#issuecomment-1568882656), [SVC](https://github.com/magento/magento-semver/pull/72) PR is also got merged.

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Semantic Version Checker, WebAPI Tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, WebAPI Tests

**Comment by engcom-Charlie** [comment]
The functional B2B and CE failures are different in recent 2 builds, seems to be flaky, not related to PR.

Functional B2b Run 1:
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32690/8702467c9bd89c3d9926ead0a52ad902/Functional/allure-report-b2b/index.html#categories
image


Run 2: 
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32690/b10797c20e9e0467f44307e30e270719/Functional/allure-report-b2b/index.html#categories/a644910ae0cb11e839a98baa21929b76/50ebaf5c92c8429c/
image


Functional CE Run 1: 
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32690/08671a99533b826f6fc78334f7ffbf3e/Functional/allure-report-ce/index.html#categories
image


Run 2: 
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32690/1b99e7cb8377f982e24ca2c0af78da92/Functional/allure-report-ce/index.html#categories
image

**Comment by engcom-Charlie** [comment]
The Web API tests seems to be a known failure. Not related to this PR. 

https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32690/062366865d06947dca33b5b66c08c41a/WebApi/console-error-logs.html
image

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
As mentioned [here](https://github.com/magento/magento2/pull/32690#issuecomment-1988202993), the semver version will get release in upcoming month, post that we can take this PR further.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
The functional EE failures in recent 2 builds on same code are not consistent. The failures are not part of PR or not failing because of the PR changes, seems to be flaky.

Run 1:
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32690/cdde4098fb4784d1f0e56281e6c531c6/Functional/allure-report-ee/index.html#categories

image


Run 2:
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32690/740ba23770eb026f3f9d87b48cee4057/Functional/allure-report-ee/index.html#categories

image

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B

**Pull request merged**
